### ISSUE TITLE
Gate.power()

### ISSUE THREAD

**Issue by ajavadia** [Qiskit/qiskit-terra#2805]
A method on the Gate class to raise it to a power.

The base implementation is easy. Corner cases:

1 -> self
0 -> Identity (I'm not sure about this as identity currently implies wait).
-1 -> inverse
-2 -> inverse().power(2)
0.5 -> square root of gate

**Comment by 1ucian0** [comment]
`power(2)` means returning a decomposable instruction, right?

**Comment by ajavadia** [comment]
> `power(2)` means returning a decomposable instruction, right?

yep.

**Comment by 1ucian0** [comment]
In that case, for the sake of type consistency, I think would be better to always return an instruction.

**Comment by 1ucian0** [comment]
Supporting rational exponentiation requires `root` method. So, if #2811 is merged, this issue will be `on hold`, blocked by #2810.
